doc/lsst.pipe.base/creating-a-pipelinetask.rst

```diff
@@ -0,0 +1,1114 @@
+                                           name="calexpBackground",
+                                           dimensions=("visit", "detector", "band"))
+        ...
+
+Now your `PipelineTask` will load the background each time the task is run.
+How do you make this optional? First, add a configuration field in your
+config class to allow the user to specify if it is to be loaded like thus
+
+.. code-block:: python
+
+    class ApertureTaskConfig(pipeBase.PipelineTaskConfig,
+                             pipelineConnections=ApertureTaskConnections):
+        ...
+        doLocalBackground = pexConfig.Field(doc="Should the background be added "
+                                                "before doing photometry",
+                                            dtype=bool, default=False)
+
+The `\_\_init\_\_` method of the connection class is given an instance of the
+tasks config class after all overrides have been applied. This provides you an
+opportunity change the behavior of the connection class according to various
+config options.
+
+.. code-block:: python
+
+    class ApertureTaskConnections(pipeBase.PipelineTaskConnections,
+                                  dimensions=("visit", "detector",
+                                              "abstract_filter", "skymap")):
+        ...
+        background = ct.Input(doc="Background model for the exposure",
+                              storageClass="Background",
+                              name="calexpBackground",
+                              dimensions=("visit", "detector", "abstract_filter",
+                                          "skymap"))
+        ...
+
+        def __init__(self, *, config=None):
+            super().__init__(config=config)
+            
+            if config.doLocalBackground is False:
+                self.inputs.remove("background")
+
+Your connection class now looks at the value of `doLocalBackground` on the
+config object and if it is false, removes it from the connection instances
+list of input connections. Connection classes keep track of what connections
+are defined in sets. Each set contains the variable names of a connection,
+and the sets themselves are identified by the type of connection they
+contain. In the example you are modifying the set of input connections. The
+names for each of the sets are as follows:
+
+* initInputs
+* initOutputs
+* inputs
+* prerequisiteInputs
+* outputs 
+
+The last step in modifying your task will be to update the run method to take
+into account that a background may or may not be supplied.
+
+.. code-block:: python
+
+    ...
+    import typing
+
+    import lsst.afw.math as afwMath
+
+    ...
+
+    class ApertureTask(pipeBase.PipelineTask):
+        
+        ...
+
+        def run(self, exposure: afwImage.Exposure,
+                inputCatalog: afwTable.SourceCatalog,
+                background: typing.Optional[afwMath.BackgroundList] = None
```

Comment:
Normally we import `Optional` directly from `typing` because it's so commonly used

```
from typing import Optional
```